openmls/src/group/tests/test_proposal_validation.rs

```diff
@@ -341,3 +341,187 @@ fn test_valsem103(ciphersuite: &'static Ciphersuite, backend: &impl OpenMlsCrypt
 
     // TODO #525: Add test for incoming proposals.
 }
+
+/// ValSem104:
+/// Add Proposal:
+/// Signature public key in proposals must be unique among existing group
+/// members
+#[apply(ciphersuites_and_backends)]
+fn test_valsem104(ciphersuite: &'static Ciphersuite, backend: &impl OpenMlsCryptoProvider) {
+    for alice_and_bob_share_keys in [
+        KeyUniqueness::NegativeSameKey,
+        KeyUniqueness::PositiveDifferentKey,
+    ] {
+        // 0. Initialize Alice and Bob
+        let alice_signature_keypair: SignatureKeypair;
+        let bob_signature_keypair: SignatureKeypair;
+
+        match alice_and_bob_share_keys {
```

Comment:
This is fine, but I think the slightly "rustier" way of doing it would be to have `match` block yield the tuple of keys. Something like:
```rust
let (alice_signature_keypair, bob_signature_keypair) = match alice_and_bob_share_keys { ... };
```